src/v/kafka/server/group.cc

```diff
@@ -1733,7 +1733,7 @@ group::store_txn_offsets(txn_offset_commit_request r) {
 }
 
 kafka::error_code map_store_offset_error_code(std::error_code ec) {
-    if (ec.category() == raft::errc_category()) {
+    if (ec.category() == raft::error_category()) {
```

Comment:
```
Fixed mapping raft error codes to kafka protocol errors. The mapping
issue was caused by incorrect use of `raft::errc_category`
ctor instead of `raft::error_category()` function.
```

Can you state in the commit message what the issue is that is being fixed?

I see that `raft::errc_category()` returns a default constructed instance, but `raft::error_category()` returns a reference to a thread local default constructed instance of `raft::errc_category` and since the comparison is equality of two references it looks like the semantics are the same. Note I think that this patch should remain, but there isn't any indication of the problem being fixed.

Reply:
`std::error_category` has overloaded `==` opearator:
```c++
       bool operator==(const error_category& __rhs) const _NOEXCEPT {return this == &__rhs;}
```
It compares addresses, this explains the bug

Reply:
p.s. i was recently reading Zig lang landing page and they claimed that operator overloading may be confusing ;)

Reply:
@mmaslankaprv wow, that's wild. thanks for the reference!